### ISSUE TITLE
some issues about GCS dataset

### ISSUE THREAD

**Issue by qwerpoiual** [InhwanBae/NPSN#5]
Thanks for your great work!
I am a little confused about the data preprocessing for GCS Dataset. In the paper, you used the first 3.2s (8 frames) as the observed trajectory and the next 4.8s (12 frames) as GT. However, I checked the paper [59], and found that the FPS is 25 and the annotated fps is 1.25. Thus, the time interval between two neighboring annotated frames is 1/1.25=0.8s. This means 8 steps in the past trajectory spend 8*0.8=6.4s, which is different from your paper. I am wondering if my understanding is correct. 
Also, could you please release the preprocessing code or provide more details about the data preprocessing for GCS dataset (train/test split)?

**Comment by InhwanBae** [comment]
Hi @qwerpoiual,

I appreciate your interest in our paper! For the GCS dataset, we performed interpolation in the same way as ETH/UCY. We also set the train/val/test to be 6:2:2. If you look at the code below, you should understand it intuitively.

```python
import os
import numpy as np

dataset_path = './GC_Dataset/Annotation/'
data_file_list = sorted(os.listdir(dataset_path))

all_data = []
for data_file in data_file_list:
    id = int(data_file.replace('.txt', ''))
    # <x_img>, <y_img>, <frame_id>
    data = np.fromfile(dataset_path + data_file, dtype=float)
    data = data.reshape((-1, 3))
    # interpolation
    data_t_interp = np.arange(np.min(data[:, 2]), np.max(data[:, 2]) + 10, 10)
    data_x_interp = np.interp(data_t_interp, data[:, 2], data[:, 0])
    data_y_interp = np.interp(data_t_interp, data[:, 2], data[:, 1])
    # <frame_id>, <object_id>, <x_coord>, <y_coord>
    data_interp = np.stack((data_t_interp, np.repeat(id, len(data_t_interp)), data_x_interp, data_y_interp))
    data_interp = data_interp.transpose()
    all_data.append(data_interp)

all_data_train = all_data[:int(len(all_data) * 0.6)]
all_data_valid = all_data[int(len(all_data) * 0.6):int(len(all_data) * 0.8)]
all_data_test = all_data[int(len(all_data) * 0.8):]
print(len(all_data), len(all_data_train), len(all_data_valid), len(all_data_test))
```

**Comment by qwerpoiual** [comment]
I've got it now. Thank you so much for taking the time to provide such a detailed response!

**Comment by qwerpoiual** [comment]
I apologize for the inconvenience, but I have another question to ask. In the GCS dataset, some trajectories of the pedestrians last more than 8 seconds. For example, a trajectory lasts 16 seconds (40 points). Do you just cut the trajectory into {[1,20], [21,40]}, or use a sliding window as {[1,20], [2,21], ..., [21,40]}? If it's convenient, could you please provide a processed data?

**Comment by InhwanBae** [comment]
Hi @qwerpoiual, We load the GCS dataset based on a sliding window with skip=1, exactly the same way as with ETH/UCY. After interpolating the data as mentioned above, we save it as a text file in the ETH/UCY format and then load it using the same dataloader. As a note, the GCS dataset is quite large, so it takes some time to load.

**Comment by qwerpoiual** [comment]
I understand. Thank you so much for your detailed response!

**Comment by HRHLALALA** [comment]
Hi @InhwanBae , my understanding of the code below is that you are spliting the train/val/test sets based on pedestrians instead of the frames? If yes, will this causes the problem when dealing with the social interaction? As some neighours in another set are not be visible?

> Hi @qwerpoiual,
> 
> I appreciate your interest in our paper! For the GCS dataset, we performed interpolation in the same way as ETH/UCY. We also set the train/val/test to be 6:2:2. If you look at the code below, you should understand it intuitively.
> 
> ```python
> import os
> import numpy as np
> 
> dataset_path = './GC_Dataset/Annotation/'
> data_file_list = sorted(os.listdir(dataset_path))
> 
> all_data = []
> for data_file in data_file_list:
>     id = int(data_file.replace('.txt', ''))
>     # <x_img>, <y_img>, <frame_id>
>     data = np.fromfile(dataset_path + data_file, dtype=float)
>     data = data.reshape((-1, 3))
>     # interpolation
>     data_t_interp = np.arange(np.min(data[:, 2]), np.max(data[:, 2]) + 10, 10)
>     data_x_interp = np.interp(data_t_interp, data[:, 2], data[:, 0])
>     data_y_interp = np.interp(data_t_interp, data[:, 2], data[:, 1])
>     # <frame_id>, <object_id>, <x_coord>, <y_coord>
>     data_interp = np.stack((data_t_interp, np.repeat(id, len(data_t_interp)), data_x_interp, data_y_interp))
>     data_interp = data_interp.transpose()
>     all_data.append(data_interp)
> 
> all_data_train = all_data[:int(len(all_data) * 0.6)]
> all_data_valid = all_data[int(len(all_data) * 0.6):int(len(all_data) * 0.8)]
> all_data_test = all_data[int(len(all_data) * 0.8):]
> print(len(all_data), len(all_data_train), len(all_data_valid), len(all_data_test))
> ```

**Comment by InhwanBae** [comment]
Hi @HRHLALALA,

You are correct. The first time I used the GCS dataset was during the DMRGCN rebuttal. Following the reviewer's request, we adopted a setup similar to the one in "Understanding Pedestrian Behaviors from Stationary Crowd Groups" (CVPR 2015), where the split is based on unseen pedestrians. Although this approach may introduce minor issues with social interaction, newly-appeared pedestrians are typically assigned a new ID, so aside from a few frames where splits occurred, we didn't observe any significant differences in experiments.
